### DISCUSSION TITLE
Teslamate disconnects each time Tesla app connects

### DISCUSSION THREAD

**Discussion by ZAQ178** [teslamate-org/teslamate#1315] [Q&A]
Each time Tesla API is logged by teslamate_1 (either by login / password or token), Tesla official iOS app shows “session expired” then automatically reconnects
Doing this, it disconnects teslamate_1 
Thus, after each time Tesla official app is opened, teslamate_1 has to be manually reconnected (interface requesting “enter login/password”) else teslamate_1 is still running but not connected nor reading any data...

(Running Teslamate on Digital Ocean VPS)
(Streaming API activated in Teslamate prefs)
(Issue appeared on Friday, both with Teslamate 1.20 & 1.20.1 and Tesla 2020.40.4 & 2020.40.8)

**Comment by toomuchlala** [comment]
Same problem here.

**Comment by jamesdeck** [comment]
Same here

**Comment by luxus** [comment]
I had the issue too.. upgraded to
Version | 1.21.0-dev from 26.10.20
Since then it looks the problem is gone.. i have 2fa on my Tesla account activated

**Comment by toomuchlala** [comment]
Upgraded. Same problem.

**Comment by robbeh1** [comment]
Same here - but it times out after about a minute even if I don’t use the Tesla App :(

**Comment by toomuchlala** [comment]
So what do we do here? Add 2FA? Rebuild our Teslamate setup? I currently use a vultr server to run docker on. I can switch to a raspberry pi if that would fix it.

**Comment by spacecosmos** [comment]
Probably something similar and/or related to this existing thread: #988

**Comment by nikobh** [comment]
Seeing something similar here. It seems that TeslaFi has a solution to the problem: https://forum.abetterrouteplanner.com/topic/2250-abrp-losing-connection-to-battery-status/?do=findComment&comment=6961

**Comment by adriankumpf** [comment]
The latest version now supports the new MFA process. Can someone try out if this solves the problem? You would have to log in again once: For this you have to delete the token in the table `tokens`. After restarting TeslaMate you will see the login form.

**Comment by spacecosmos** [comment]
> The latest version now supports the new MFA process. Can someone try out if this solves the problem? You would have to log in again once: For this you have to delete the token in the table `tokens`. After restarting TeslaMate you will see the login form.

Is there any way you can maybe add a "remove existing token" option from the settings page? That way, people can easily renew the token whenever needed. Instead of messing in database tables on their own. 😉

**Comment by airmaxx** [comment]
I removed the token, switched to MFA, yet the error messages still persist.

e.g.: 
> 2020-12-30 11:05:44.801 [warn] Too many disconnects …
2020-12-30 11:05:44.801 car_id=2 [info] Creating new connection …
2020-12-30 11:05:44.801 car_id=2 [info] Disconnecting ...
2020-12-30 11:05:44.806 car_id=2 [info] Connecting ...
2020-12-30 11:05:55.729 [info] Connection was closed (a:1|t:1|d:0). Reconnecting …
2020-12-30 11:06:06.653 [info] Connection was closed (a:1|t:2|d:0). Reconnecting …
2020-12-30 11:06:17.566 [info] Connection was closed (a:1|t:3|d:0). Reconnecting …
2020-12-30 11:06:28.496 [info] Connection was closed (a:1|t:4|d:0). Reconnecting …

**Comment by tohojo** [comment]
airmaxx <notifications@github.com> writes:

> I removed the token, switched to MFA, yet the error messages still persist.
>
> e.g.: 
>> 2020-12-30 11:05:44.801 [warn] Too many disconnects …
> 2020-12-30 11:05:44.801 car_id=2 [info] Creating new connection …
> 2020-12-30 11:05:44.801 car_id=2 [info] Disconnecting ...
> 2020-12-30 11:05:44.806 car_id=2 [info] Connecting ...
> 2020-12-30 11:05:55.729 [info] Connection was closed (a:1|t:1|d:0). Reconnecting …
> 2020-12-30 11:06:06.653 [info] Connection was closed (a:1|t:2|d:0). Reconnecting …
> 2020-12-30 11:06:17.566 [info] Connection was closed (a:1|t:3|d:0). Reconnecting …
> 2020-12-30 11:06:28.496 [info] Connection was closed (a:1|t:4|d:0).
> Reconnecting …

I also still get errors; including some new ones:

```
Dec 30 11:08:16 teslamate teslamate[13835]: 2020-12-30 12:08:16.551 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …
Dec 30 11:08:27 teslamate teslamate[13835]: 2020-12-30 12:08:27.386 [info] Connection was closed (a:1|t:2|d:1). Reconnecting …
Dec 30 11:08:52 teslamate teslamate[13835]: 2020-12-30 12:08:52.917 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …
Dec 30 11:09:03 teslamate teslamate[13835]: 2020-12-30 12:09:03.755 [info] Connection was closed (a:1|t:2|d:1). Reconnecting …
Dec 30 11:09:11 teslamate teslamate[13835]: 2020-12-30 12:09:11.313 car_id=1 [warn] Discarded incomplete fetch result
Dec 30 11:09:14 teslamate teslamate[13835]: 2020-12-30 12:09:14.526 [info] Connection was closed (a:1|t:3|d:1). Reconnecting …
Dec 30 11:09:21 teslamate teslamate[13835]: 2020-12-30 12:09:21.765 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:09:21 teslamate teslamate[13835]: 2020-12-30 12:09:21.765 car_id=1 [error] Error / :unknown
Dec 30 11:09:25 teslamate teslamate[13835]: 2020-12-30 12:09:25.301 [info] Connection was closed (a:1|t:4|d:1). Reconnecting …
Dec 30 11:09:42 teslamate teslamate[13835]: 2020-12-30 12:09:42.263 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:09:42 teslamate teslamate[13835]: 2020-12-30 12:09:42.263 car_id=1 [error] Error / :unknown
Dec 30 11:10:02 teslamate teslamate[13835]: 2020-12-30 12:10:02.751 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:10:02 teslamate teslamate[13835]: 2020-12-30 12:10:02.752 car_id=1 [error] Error / :unknown
Dec 30 11:10:23 teslamate teslamate[13835]: 2020-12-30 12:10:23.208 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:10:23 teslamate teslamate[13835]: 2020-12-30 12:10:23.208 car_id=1 [error] Error / :unknown
Dec 30 11:10:44 teslamate teslamate[13835]: 2020-12-30 12:10:44.015 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:10:44 teslamate teslamate[13835]: 2020-12-30 12:10:44.015 car_id=1 [error] Error / :unknown
Dec 30 11:10:54 teslamate teslamate[13835]: 2020-12-30 12:10:54.573 [info]     :alarm_handler: {:set, {TeslaMate.Vehicles.Vehicle_1_api_error, :fuse_blown}}
Dec 30 11:11:04 teslamate teslamate[13835]: 2020-12-30 12:11:04.578 [warn] TeslaApi.Error / upstream internal error
Dec 30 11:11:04 teslamate teslamate[13835]: 2020-12-30 12:11:04.578 car_id=1 [error] Error / :unknown
Dec 30 11:11:27 teslamate teslamate[13835]: 2020-12-30 12:11:27.390 car_id=1 [info] Suspending logging
```

**Comment by ZAQ178** [comment]
Not tried yet
Nevertheless  I bypassed the issue by creating a special  user / login dedicated to Teslamate, as a "virtual" additional driver
I thus have 2 different profiles : 
- xxx@gmail.com + password1 for official Tesla App
- yyy@zzz.com + password2 for Teslamate and/or associated token
No MFA activated
This way, no errors !?

**Comment by airmaxx** [comment]
I now also have a second virtual user running just for Teslamate. MFA is also activated there. Unfortunately, the disconnects occur just as before. Interestingly, my second logger is not affected by this.
